src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +565,69 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static bool ValidateArguments(string[] args, CommandLineApplication app)
+    {
+        static HashSet<string> GetValidArguments(CommandLineApplication app)
+        {
+            var validArguments = new HashSet<string>();
+            foreach (var option in app.GetOptions())
+            {
+                if (!string.IsNullOrEmpty(option.LongName))
+                {
+                    validArguments.Add(option.LongName);
+                }
+                if (!string.IsNullOrEmpty(option.ShortName))
+                {
+                    validArguments.Add(option.ShortName);
+                }
+            }
+
+            return validArguments;
+        }
+
+        static ReadOnlyMemory<char> GetArgumentName(string arg) => arg.StartsWith("--") ? arg.AsMemory(2) : arg.StartsWith('-') ? arg.AsMemory(1) : ReadOnlyMemory<char>.Empty;
+        static IEnumerable<ReadOnlyMemory<char>> GetArgumentNames(IEnumerable<string> args)
+        {
+            bool lastWasArgument = false;
+            foreach (ReadOnlyMemory<char> potentialArgument in args.Select(GetArgumentName))
+            {
+                if (!lastWasArgument)
+                {
+                    bool isCurrentArgument = lastWasArgument = !potentialArgument.IsEmpty;
+                    if (isCurrentArgument)
+                    {
+                        yield return potentialArgument;
+                    }
+                }
+                else
+                {
+                    lastWasArgument = false;
+                }
+            }
+        }
+
+        // Get all valid options from the configuration files
+        var validArguments = GetValidArguments(app);
+
+        var argumentsNamesProvided = GetArgumentNames(args).ToList();
+        foreach (var argumentName in argumentsNamesProvided)
+        {
+            // Check if the argument provided is a valid option/argument
+            if (!validArguments.Contains(argumentName.ToString()))
```

Comment:
Ah we need that `ToString` conversion here...
But maybe we can made `HashSet<ReadOnlyMemory<char>>` to avoid it?

Reply:
I did try but it didn't work the way I did it yesterday, however, today I realized that I had to tweak the implementation of `GetValidArguments` a little bit to make it work, and it works now